src/main/webapp/script.js

```diff
@@ -71,22 +72,66 @@ function passQuery() {
     });
 }
 
-function displayAmChartsMap(data, description, location, isCountyQuery) {
-  am4core.useTheme(am4themes_animated);
-  const chart = am4core.create('map', am4maps.MapChart);
-  chart.height = 550;
-  chart.zoomControl = new am4maps.ZoomControl();
-  // only allow zooming with buttons
-  chart.mouseWheelBehavior = 'none';
-
-  // Create map instance
+function getGeoData(location, isCountyQuery) {
   if (isCountyQuery) {
-    chart.geodata = location ===
+    return location ===
         '06' ? am4geodata_region_usa_caLow : am4geodata_region_usa_njLow;
   } else {
-    chart.geodata = am4geodata_usaLow;
+    return am4geodata_usaLow;
   }
+}
+
+
+function displayVisualization(censusDataArray, description,
+  location, isCountyQuery) {
+  const geoData = getGeoData(location, isCountyQuery);
+  if (isCountyQuery) {
+    setStyleForCountyQuery();
+    const mapsData = getMapsData(censusDataArray);
+    const amChartsData = createDataArray(censusDataArray, isCountyQuery);
+    displayAmChartsMap(amChartsData, description, geoData);
+    displayCountyGeoJson(mapsData, location);
+  } else {
+    setStyleForStateQuery();
+    const amChartsData = createDataArray(censusDataArray, isCountyQuery);
+    displayAmChartsMap(amChartsData, description, geoData);
+  }
+  document.getElementById('more-info').innerText = '';
+}
 
+// Check that the input being written to a datalist can match one of its options
```

Comment:
This is a more general comment, but we now have quite a lot of different functions in this JS file. I think it might make sense to create some "sections" within the file (just by using comments) for various types of functions, so that input validation isn't all mixed up with data visualization, and so on.

Reply:
Very true! Will merge and then we can work on breaking it up.